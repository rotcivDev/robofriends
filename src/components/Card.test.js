import React from 'react';
import Card from './Card';
import {render, cleanup, screen} from '@testing-library/react';

describe('Card Component', () => {
  afterEach(cleanup)

  test('should render layout correctely passing name, email and id props dynamically', async () => {

    const mockProps = {
      name: 'Dynamic name',
      email: 'vsallesa@gmail.com',
      id: '10'
    }
    
    expect(render(<Card name={mockProps.name} email={mockProps.email} id={mockProps.id}/>)).toMatchSnapshot();
  
    const dinamicName = await screen.findByText(mockProps.name);
    const dinamicEmail = await screen.findByText(mockProps.email);
    const cardImg = await screen.findByAltText('robots');
    
    expect(cardImg).toBeInTheDocument()
    expect(dinamicName).toBeInTheDocument()
    expect(dinamicEmail).toBeInTheDocument()
  })

})
