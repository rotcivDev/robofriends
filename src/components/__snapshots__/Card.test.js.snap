// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render layout correctely passing name, email and id props dynamically 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
      >
        <img
          alt="robots"
          src="https://robohash.org/10?size=200x200"
        />
        <div>
          <h2>
            Dynamic name
          </h2>
          <p>
            vsallesa@gmail.com
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
    >
      <img
        alt="robots"
        src="https://robohash.org/10?size=200x200"
      />
      <div>
        <h2>
          Dynamic name
        </h2>
        <p>
          vsallesa@gmail.com
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
